File: 03-spring-examples/apps/utils/src/main/java/com/nalepa/demo/common/monitored/customSimpleMonitoredExecutor/MonitoredExecutor.java
package com.nalepa.demo.common.monitored.customSimpleMonitoredExecutor;

import io.micrometer.core.instrument.MeterRegistry;
import java.util.concurrent.Executor;

// add here metric about tasks in queue?
public class MonitoredExecutor implements Executor {

    private final Executor delegate;
    public final String messagePrefix;
    public final String metricName;
    public final MeterRegistry meterRegistry;

    public MonitoredExecutor(Executor delegate, String messagePrefix, String metricName, MeterRegistry meterRegistry) {
        this.delegate = delegate;
        this.messagePrefix = messagePrefix;
        this.metricName = metricName;
        this.meterRegistry = meterRegistry;
    }

    @Override
    public void execute(Runnable runnable) {
        delegate.execute(new MonitoredRunnable(runnable, messagePrefix, metricName, meterRegistry));
    }
}

File: apps/commons/src/main/kotlin/com/nalepa/demo/common/monitored/customMonitoredExecutor/MonitoredExecutor.kt
package com.nalepa.demo.common.monitored.customMonitoredExecutor

import io.micrometer.core.instrument.MeterRegistry
import java.util.concurrent.Executor

// add here metric about tasks in queue?
class MonitoredExecutor(
    private val delegate: Executor,
    val messagePrefix: String,
    val metricName: String,
    val meterRegistry: MeterRegistry,
) : Executor {
    override fun execute(runnable: Runnable) {
        delegate.execute(MonitoredRunnable(runnable, messagePrefix, metricName, meterRegistry))
    }
}

